purchases/src/test/java/com/revenuecat/purchases/attributes/SubscriberAttributesDeviceCacheTests.kt

```diff
@@ -0,0 +1,220 @@
+package com.revenuecat.purchases.attributes
+
+import android.content.SharedPreferences
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.revenuecat.purchases.caching.DeviceCache
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.runs
+import io.mockk.slot
+import org.assertj.core.api.Assertions.assertThat
+import org.json.JSONObject
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.Date
+
+@RunWith(AndroidJUnit4::class)
+class SubscriberAttributesDeviceCacheTests {
+    private val apiKey = "api_key"
+    private val appUserID = "app_user_id"
+
+    private lateinit var underTest: DeviceCache
+    private lateinit var mockPrefs: SharedPreferences
+    private lateinit var mockEditor: SharedPreferences.Editor
+    private val putStringSlot = slot<String>()
+
+    @Before
+    fun setup() {
+        mockPrefs = mockk()
+        mockEditor = mockk()
+        every {
+            mockEditor.putString(any(), capture(putStringSlot))
+        } returns mockEditor
+        every {
```

Comment:
maybe add a line break between these two lines for consistency with the others

Reply:
I refactored this a little bit